File: Extracted.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;
using System.Web;

public static class Test
{
    public static string Decrypt(string value)
    {
        string keyString = value.Substring(0, 9);
        string encryptedEncodedString = value.Substring(9);
        
        int calculatedIndex = 0;
        int index = 0;

        // #region Decoding
        if ((encryptedEncodedString = Regex.Replace(encryptedEncodedString, "[ \t\n\f\r]", "")).Length % 4 == 0)
        {
            encryptedEncodedString = Regex.Replace(encryptedEncodedString, "==?$", "");
            if (encryptedEncodedString.Length % 4 == 1 || new Regex("[^+/0-9A-Za-z]").IsMatch(encryptedEncodedString))
            {
                return null;
            }
        }

        string encryptedDecodedString = "";

        int offset = 0;
        for (index = 0; index < encryptedEncodedString.Length; index++)
        {
            offset <<= 6;

            int charIndex = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/".IndexOf(encryptedEncodedString[index]);
            offset |= charIndex < 0 ? 0 : charIndex;

            if (((index + 1) % 4) == 0)
            {
                encryptedDecodedString += Convert.ToChar((16711680 & offset) >> 16);
                encryptedDecodedString += Convert.ToChar((65280 & offset) >> 8);
                encryptedDecodedString += Convert.ToChar((255 & offset));
                offset = 0;
            }
            if (index == encryptedEncodedString.Length - 1)
            {
                switch (index % 4)
                {
                    case 1:
                        offset >>= 4;
                        encryptedDecodedString += Convert.ToChar(offset);
                        break;
                    case 2:
                        offset >>= 2;
                        encryptedDecodedString += Convert.ToChar((65280 & offset) >> 8);
                        encryptedDecodedString += Convert.ToChar(((255) & offset));
                        break;
                }
            }
        }

        try
        {
            encryptedDecodedString = HttpUtility.UrlDecode(encryptedDecodedString);
        }
        catch (Exception err) { }
        // #endregion

        // #region Decryption
        Dictionary<int, int> tmpArray = Enumerable.Range(0, 256).Select(i => new { Key = i, Value = i }).ToDictionary(x => x.Key, x => x.Value);

        for (index = 0; index < 256; index++)
        {
            calculatedIndex = ((calculatedIndex + tmpArray[index] + Convert.ToUInt16(keyString.ToCharArray()[index % keyString.Length])) % 256);

            int tmp = tmpArray[index];
            tmpArray[index] = tmpArray[calculatedIndex];
            tmpArray[calculatedIndex] = tmp;
        }

        string decryptedDecodedString = "";
        for (int s = calculatedIndex = index = 0; s < encryptedDecodedString.Length; s++)
        {
            calculatedIndex = (calculatedIndex + tmpArray[index = (index + 1) % 256]) % 256;

            int tmp = tmpArray[index];
            tmpArray[index] = tmpArray[calculatedIndex];
            tmpArray[calculatedIndex] = tmp;

            decryptedDecodedString += Convert.ToChar(Convert.ToUInt16(encryptedDecodedString.ToCharArray()[s]) ^ tmpArray[(tmpArray[index] + tmpArray[calculatedIndex]) % 256]);
        }
        // #endregion

        return decryptedDecodedString;
    }

    public static void Main()
    {
        Console.Clear();
        Console.Write("Enter Encrypted String\t:\t");
        string value = Console.ReadLine();
        Console.WriteLine("Decrypted String\t:\t{0}",Decrypt(value));
    }
}

File: Extracted.js
var Keys = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"

function Decrypt(value) {
    var n = value.substr(0, 9)
        , i = f1(value.substr(9));
    try {
        i = decodeURIComponent(i)
    } catch (err) { }
    return f2(n, i)
}

function f1(value) {
    if ((value = (value = "".concat(value)).replace(/[ \t\n\f\r]/g, "")).length % 4 == 0 && (value = value.replace(/==?$/, "")),
        value.length % 4 == 1 || /[^+/0-9A-Za-z]/.test(value))
        return null;
    for (var x, r, u = "", c = 0, e = 0, f = 0; f < value.length; f++)
        c <<= 6,
            c |= (x = value[f],
                r = void (0),
                (r = Keys.indexOf(x)) < 0 ? void (0) : r),
            24 === (e += 6) && (u += String.fromCharCode((16711680 & c) >> 16),
                u += String.fromCharCode((65280 & c) >> 8),
                u += String.fromCharCode((255 & c)),
                c = e = 0);
    return 12 === e ? (c >>= 4,
        u += String.fromCharCode(c)) : 18 === e && (c >>= 2,
            u += String.fromCharCode((65280 & c) >> 8),
            u += String.fromCharCode(((255) & c))),
        u
}

function f2(t, n) {
    for (var x, r = [], u = 0, c = "", e = 256, f =0; f < e; f += 1)
        r[f] = f;
    for (f = 0; f < e; f += 1)
        u = ((u + r[f] + t.charCodeAt(f % t.length)) % e),
            x = r[f],
            r[f] = r[u],
            r[u] = x;
    for (var s = u = f = 0; s < n.length; s += 1)
        u = (u + r[f = (f + 1) % e]) % e,
            x = r[f],
            r[f] = r[u],
            r[u] = x,
            c += String.fromCharCode(n.charCodeAt(s) ^ r[(r[f] + r[u]) % e]);
    return c
}
